### PR TITLE
check the permission to access localStorage (fix #5)

### LINKED ISSUE #5: Uncaught SecurityError when cookie settings block localStorage
Chrome throws a SecurityError when accessing ``window.localStorage`` with [cookie settings](https://support.google.com/chrome/answer/95647?hl=en&co=GENIE.Platform=Desktop) set to *"Block sites from setting any data"*:
``Uncaught SecurityError: Failed to read the 'localStorage' property from 'Window': Access is denied for this document.`` in ``leaflet.restoreview.js:3``

The Pale Moon browser (and probably older Firefox) is also throwing a SecurityError with restrictive cookie settings.

See also:
* [MDN: Using the Web Storage API - Testing for support vs availability](https://developer.mozilla.org/en-US/docs/Web/API/Web_Storage_API/Using_the_Web_Storage_API#Testing_for_support_vs_availability)
* [Modernizr localstorage.js](https://github.com/Modernizr/Modernizr/blob/master/feature-detects/storage/localstorage.js) feature detect

### PR DESCRIPTION
**Pull request by enguerran**
It could fix #5

cf. https://developer.mozilla.org/en-US/docs/Web/API/Web_Storage_API/Using_the_Web_Storage_API\#Testing_for_support_vs_availability

r?

### CODE REVIEW

File: leaflet.restoreview.js [hunk 1]
```diff
@@ -1,31 +1,49 @@
...
-            this.setView(L.latLng(view.lat, view.lng), view.zoom, true);
+            var storage = window[type],
+                x = '__storage_test__';
+            storage.setItem(x, x);
+            storage.removeItem(x);
             return true;
         }
-        catch (err) {
+        catch(e) {
+            console.warn("Your browser block access to " + type);
```
Comment (leplatrem) [inline leaflet.restoreview.js]: *blocks*

File: leaflet.restoreview.js [hunk 2]
```diff
@@ -1,31 +1,49 @@
...
-        var storage = window.localStorage || {};
-        if (!this.__initRestore) {
-            this.on('moveend', function (e) {
-                if (!this._loaded)
-                    return;  // Never access map bounds if view is not set.
+(function() {
+    var RestoreViewMixin = {
+        restoreView: function () {
+            if (storageAvailable('localStorage')) {
+                var storage = window.localStorage || {};
```
Comment (leplatrem) [inline leaflet.restoreview.js]: the `|| {}` should now be superfluous IMO

File: leaflet.restoreview.js [hunk 3]
```diff
@@ -1,31 +1,49 @@
...
-    restoreView: function () {
-        var storage = window.localStorage || {};
-        if (!this.__initRestore) {
-            this.on('moveend', function (e) {
-                if (!this._loaded)
-                    return;  // Never access map bounds if view is not set.
+(function() {
+    var RestoreViewMixin = {
+        restoreView: function () {
+            if (storageAvailable('localStorage')) {
```
Comment (leplatrem) [inline leaflet.restoreview.js]: Instead of enclosing the whole block of code within this `if`, why not a quick return if not available?

**Comment by enguerran** [comment]
I add a note in the readme [as requested](https://github.com/makinacorpus/Leaflet.RestoreView/issues/5#issuecomment-252019017)

**Pull request merged**